### PR TITLE
Flutter challenge

### PR DESCRIPTION
**Pull request by isacmoura**

### CODE REVIEW

**Review by hrbocutti — APPROVED** [review]
Good skills and approach on code.
Congrats.

**Comment by joaomantovani** [comment]
Nice working!!

You've used some packages that we use here in penseapp, like Bloc and Hive. Which does not interfere on test, but it's nice!

- [x] You've used the `repository` pattern
- [x] You've used exceptions to handle success/failure
- [x] You've used Models
- [x] Separated the folders on a good approach
- [x] Bloc/Cubit
- [x] Nice readme
- [x] Splash

On the general, it's a very good project, congrats.

Some hints to you that do not interfere on the test.

Here in penseapp, we use two packages to handle some code generation when working with Bloc/Models, which save us a lot of time.

- https://pub.dev/packages/freezed
- https://pub.dev/packages/json_serializable

Specially in Bloc, the freezed package generates a lot of boilerplate which save us a lot of time! But as I mentioned before, this is  just a hint, not that interfere on test!

**Review by joaomantovani — APPROVED** [review]

**Comment by bernesdev** [comment]
- Great Documentation
- Good Project Structure (It's ease to understand what each thing does)
- Use of Bloc/Cubit 
- The Session is Saved
- Error/Info Alerts
- Good Splash Screen

Some problems (nothing big):
- Could have a success message when the new account is created
- It's possible to add the same items to the wishlist multiple times
- I could not find the downloaded list
- It would be good to have a visual feedback when an item is added to the wishlist, like a "check" icon

In summary you did a good job, congratulations. 😃👏
